### PR TITLE
removed tag from user table

### PR DESCRIPTION
**Pull request by Retler**
### Description
We are removing the redundant tag from the user table.

### Parent Issue
Issue #92 

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [x] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### Testing Checklist
- [x] I have added/updated unit tests for functional changes.
- [x] I have added/updated integration tests for changes that affect dependent systems.
- [x] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [ ] I have updated relevant workflows and deployment tools.
- [ ] I have updated/added relevant documentation (readme, doc comments, etc).
- [ ] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: encryption-service/authstorage/authstorage.go [hunk 1]
```diff
@@ -165,26 +165,26 @@ func (storeTx *AuthStoreTx) NewQuery(query string) string {
...
+	row := storeTx.tx.QueryRow(ctx, storeTx.NewQuery("SELECT * FROM users WHERE id = $1"), userID)
+	err := row.Scan(&fetchedID)
 	if errors.Is(err, pgx.ErrNoRows) {
 		return nil, ErrNoRows
 	}
 	if err != nil {
 		return nil, err
 	}
-	return storedTag, nil
+	return fetchedID, nil
```
[the remark this answers is not in the document]
Comment (iead-cc) [reply encryption-service/authstorage/authstorage.go:181]: The result is also discarded in the only place this function is called. Only the error is relevant

[the remark this answers is not in the document]
Comment (Retler) [reply encryption-service/authstorage/authstorage.go]: I have change GetUser to UserExists

File: encryption-service/app/permissions_handler.go [hunk 2]
```diff
@@ -84,15 +84,18 @@ func (app *App) AddPermission(ctx context.Context, request *AddPermissionRequest
...
-	_, err = authStorageTx.GetUser(ctx, target)
+	exists, err := authStorageTx.UserExists(ctx, target)
+
+	if !exists {
+		msg := fmt.Sprintf("AddPermission: Failed to retrieve target user %v", target)
+		err = status.Errorf(codes.InvalidArgument, "invalid target user ID")
+		log.Error(ctx, msg, err)
+		return nil, err
+	}
 	if err != nil {
```
Comment (iead-cc) [inline encryption-service/app/permissions_handler.go:95]: I think the error should be checked first

Comment (Retler) [reply encryption-service/app/permissions_handler.go:95]: Ah yes, very good point!

**Review by iead-cc — APPROVED** [review]
lgtm

**Pull request merged**